### PR TITLE
[17.0][MIG] sale_substate

### PR DESCRIPTION
**Pull request by Wodran14**
Depends on:

- [https://github.com/OCA/server-ux/pull/855]

### CODE REVIEW

File: sale_substate/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        selection_add=[("sale.order", "Sale order")], ondelete={"sale.order": "cascade"}
+    )
+
+
+class SaleOrder(models.Model):
+    _inherit = ["sale.order", "base.substate.mixin"]
+    _name = "sale.order"
+
+    @api.constrains("substate_id", "state")
+    def check_substate_id_value(self):
```
Comment (rousseldenis) [inline sale_substate/models/sale_order.py:21]: @Wodran14 IMHO this is covered by : https://github.com/OCA/server-ux/pull/855/files#diff-84ca126b7a08243bd6268af8f3c6e2ffe5ffa1e2d82b9cdb867840cd14a0b2f4R14

File: sale_substate/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                        " %(state)s but for %(target_state)s "
+                    )
+                    % {
+                        "name": order.substate_id.name,
+                        "state": _(sale_states[order.state]),
+                        "target_state": _(sale_states[target_state]),
+                    }
+                )
+
+    def _track_template(self, changes):
```
Comment (rousseldenis) [inline sale_substate/models/sale_order.py:38]: And this one by : https://github.com/OCA/server-ux/pull/855/files#diff-84ca126b7a08243bd6268af8f3c6e2ffe5ffa1e2d82b9cdb867840cd14a0b2f4R31

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: sale_substate/views/sale_views.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    <record id="sale_order_view_form" model="ir.ui.view">
+      <field name="model">sale.order</field>
+      <field name="inherit_id" ref="sale.view_order_form" />
+      <field name="arch" type="xml">
+          <xpath expr="//header" position="after">
+              <header name="substate">
+                  <field
+                        name="substate_id"
+                        widget="statusbar"
+                        options="{'clickable': '1',}"
```
Comment (ellbristow) [inline sale_substate/views/sale_views.xml:14]: Suggestion: Hide states that do not apply to the current parent state

```suggestion
                        options="{'clickable': '1',}"
                        domain="[('target_state_value_id.target_state_value', '=', state)]"
```

**Review by ellbristow — COMMENTED** [review]
A suggestion. Something I have had to implement for two separate clients

**Comment by rousseldenis** [comment]
@Wodran14

**Comment by ferran-S73** [comment]
Functional review OK

**Comment by CRogos** [comment]
@Wodran14 are you still working on this PR?
Dependency is now available: https://github.com/OCA/server-ux/pull/940

**Comment by flotho** [comment]
ping @Wodran14 
any change to have a rebase so CI will be triggered

**Comment by MohamedOsman7** [comment]
> ping @Wodran14 any change to have a rebase so CI will be triggered

i opened a new PR for this modul, could you make a review? #3418

**Comment by rousseldenis** [comment]
Closing as no answer from author. See #3418

**Pull request closed without merging**